src/node/db/SecurityManager.js

```diff
@@ -70,6 +71,13 @@ exports.checkAccess = async (padID, sessionCookie, token, userSettings) => {
     // Note: userSettings.padAuthorizations should still be populated even if
     // settings.requireAuthorization is false.
     const padAuthzs = userSettings.padAuthorizations || {};
+    if (readOnlyManager.isReadOnlyId(padID)) {
+      // pad is read-only, first get the real pad ID; some callers of checkAccess() first fetch the real pad ID,
+      // but some do not
+      padID = await readOnlyManager.getPadId(padID);
```

Comment:
The alternative to performing normalization here would be ensuring that securityManager.checkAccess() is never called with read-only pad IDs. As far as I can tell, this would currently only affect src/node/padaccess.js (which does not get the real pad ID before querying checkAccess) which is used by pad export.
However, I have decided to perform normalization here in order to prevent similar bugs in the future if developers would forget to "normalize" the pad ID elsewhere.